datalad/support/gitrepo.py

```diff
@@ -1616,14 +1685,18 @@ def get_files(self, branch=None):
         [str]
           list of files.
         """
-        # TODO: RF codes base and melt get_indexed_files() in
-
-        if branch is None:
-            # active branch can be queried way faster:
-            return self.get_indexed_files()
-        else:
-            return [item.path for item in self.repo.tree(branch).traverse()
-                    if isinstance(item, Blob)]
+        branch = 'HEAD' if branch is None else branch
+        tree = '' if tree is None else tree
+        files = []
+        for f in self.get_object('{}:{}'.format(
+                branch, tree)):
+            if f.endswith('/'):
```

Comment:
Wouldn't a single `ls-tree -r` call be more efficient?

Reply:
Absolutely. I was more or less playing around in the scope of get_object(). I don't need this function at all. -revolution has get_content_info(), which is built on ls-tree and ls-files, but we cannot have it, because -core must support direct mode users  -- deprecated in git-annex, but prime feature of datalad :(

I'll probably just revert this part of the diff.